src/main/java/esa/sen2vm/input/gipp/GIPPFileManager.java

```diff
@@ -0,0 +1,169 @@
+
+    /**
+     * List of GIPP Viewing Direction XML files
+     */
+    protected List<File> viewingDirectionFileList = new ArrayList<>();
+
+    /**
+     * GIPP Blind pixel XML files
+     */
+    protected File blindPixelFile = null;
+
+    /**
+     * GIPP spa mod XML files
+     */
+    protected File spaModFile = null;
+
+    /**
+     * Default constructor
+     */
+    public GIPPFileManager() {
+
+    }
+
+    /**
+     * Get GIPP file of a specific type by passing a regex
+     * @param folders that may
+     * @param a regex that match a specific type of GIPP
+     * @return an array that contains all files that correspond to the input regex
+     */
+    public List<File> findFiles(File[] directories, String regexMatch) {
+        List<File> foundFiles = new ArrayList<>();
+        Pattern pattern = Pattern.compile(regexMatch);
+
+        if (directories != null) {
+            for (File dir : directories) {
+                if (dir.isDirectory()) {
+                    File[] files = dir.listFiles(new FilenameFilter() {
+                        @Override
+                        public boolean accept(File dir, String name) {
+                            Matcher matcher = pattern.matcher(name);
+                            return matcher.matches();
+                        }
+                    });
+
+                    if (files != null) {
+                        for (File file : files) {
+                            foundFiles.add(file);
+                        }
+                    }
+                }
+            }
+        }
+
+        if (foundFiles != null && foundFiles.size() > 0) {
+            return foundFiles;
+        }
+        else {
+            return null;
+        }
+    }
+
+    /**
+     * Get GIPP file of a specific type by passing a regex
+     * @param folder contains the GIPP xml files
+     * @param a regex that match a specific type of GIPP
+     * @return the first occurrence that correspond to the input regex
+     */
+    public File findFile(File[] directories, String regexMatch) {
+        List<File> foundFiles = findFiles(directories, regexMatch);
+
+        if (foundFiles != null && foundFiles.size() > 0) {
+            return foundFiles.get(0);
+        }
+        else {
+            return null;
+        }
+    }
+
+    /**
+     * Constructor
+     * @param folder contains the GIPP xml files
+     */
+    public GIPPFileManager(String folder) {
+        File gippFolder = new File(folder);
+
+        if (!gippFolder.exists() || !gippFolder.isDirectory()) {
+            LOGGER.severe("The gipp folder " + gippFolder + " doesn't exist or is not a directory");
+        }
+
+        File[] directories = gippFolder.listFiles();
+
+        // get viewing direction files
+        viewingDirectionFileList = findFiles(directories, ".*GIP_VIEDIR.*\\.DBL");
+
+        // get blind pixel file
+        blindPixelFile = findFile(directories, ".*GIP_BLINDP_MPC.*\\.DBL");
```

Comment:
Put values in constants (GIP_BLINDP & .DLB)
Remove _MPC from the end of the GIP_BLINDP filter